### PR TITLE
[FE] Remove [Rule/Policy]Summary and [Rule/Policy]Details from GraphQL

### PR DESCRIPTION
**Pull request by 3nvi**
## Background

This PR is a pre-cursor to unified detections functionality, in which we cleanup our GraphQL types in order to make them more explicit. Specifically:

- `RuleDetails` and `RuleSummary` have been replaced by `Rule`
- `PolicyDetails` and `PolicySummary` have been replaced by `Policy`

This will greatly help us during updates & deletions since we no longer need to make multiple updates into different GraphQL types. It also helps us create (eventually) a single `Detection` type that can be `Rule | Policy`.

Relates to #1821 

## Changes

- Remove all `[Rule/Policy]`Summary and `[Rule/Policy]Details` references and replace with `[Rule/Policy]`
- Cleanup un-needed Fragments and rename them in order to use the **newly freed** `xxxSummary` and `xxxDetails` names. This means that now, whenever we see a `xxxSummary` and `xxxDetails` type, it's always gonna be a fragment on `xxx` (i.e. `RuleSummary` is a fragment on `Rule`)
- Update tests since our new types changed some fixture values

## Notes

This migration is possible since the lambda running the `analysis-api` returns the full Policy & Rule objects in all exposed  functions. This allows us to do the same as well. 

This wasn' t the case a while ago, but it now is.

## Testing

- `npm run test` - the fact that automated tests & `tsc` succeed means  that the migration was smooth
- Manual tests

### CODE REVIEW

**Review by alexmylonas — COMMENTED** [review]
Great job Aggelos, that must have been tedius. 

Some suggestions: 
- Rename `ruleTeaser.graphql` to `getRuleSummary.graphql` and remove & replace `getRuleDetails.graphql` in `RuleDetails` page

-

File: web/src/pages/EditRule/graphql/getRuleDetails.graphql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-query RuleDetails($input: GetRuleInput!) {
+query GetRuleDetails($input: GetRuleInput!) {
   rule(input: $input) {
     body
     dedupPeriodMinutes
```
[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/pages/EditRule/graphql/getRuleDetails.graphql]: Yup, nice idea. Done!

File: web/src/pages/EditRule/graphql/getRuleDetails.graphql [hunk 2]
```diff
@@ -1,4 +1,4 @@
-query RuleDetails($input: GetRuleInput!) {
+query GetRuleDetails($input: GetRuleInput!) {
```
[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/pages/EditRule/graphql/getRuleDetails.graphql:1]: Wow, you are right! Made sure that both return `...RuleDetails`

File: web/src/pages/AlertDetails/PolicyAlertDetails/graphql/policyTeaser.graphql [hunk 3]
```diff
@@ -1,5 +1,5 @@
 query PolicyTeaser($input: GetPolicyInput!) {
```
[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/pages/AlertDetails/PolicyAlertDetails/graphql/policyTeaser.graphql]: Nice catch! Done!

**Review by alexmylonas — APPROVED** [review]
LGTM!

**Pull request merged**